### PR TITLE
617/618 Find Spend Activities Front and Back

### PR DESCRIPTION
**Pull request by dupreesi**
#### Issues this PR relates to

---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist
Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)
- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: server/modules/activity/model/find.js [hunk 1]
```diff
@@ -3,30 +3,29 @@ import { query } from '../../../database';
...
+                  sa.activity_types::VARCHAR[] "activity_types",
+                  sa.is_online "online",
+                  sp.city "townOrCity",
+                  i.file_name "image_name",
+                  i.key "image_key",
+                  i.bucket "image_bucket",
+                  i.bucket_region "image_bucket_region",
                   COALESCE(sp.name, eap.name) "spendVenue.name",
-                  COALESCE(sp.user_id, eap.user_id) "spendVenue.id"
+                  COALESCE(sp.user_id, eap.user_id) "spendVenue.id",
```
Comment (RamyAlshurafa) [inline server/modules/activity/model/find.js:18]: `"spendVenue.id"` => `"spend_venue.id"` to captalize it 
same for all `spendVenues.xxx`

File: server/modules/activity/model/find.js [hunk 2]
```diff
@@ -3,30 +3,29 @@ import { query } from '../../../database';
...
+                  (SELECT COUNT(sap.id)
+                     FROM spend_activity_participants sap
+                     WHERE sa.id = sap.spend_activity_id
+                     ORDER BY COUNT(sap.id) DESC
+                     ) AS "no_transactions"
                 FROM spend_activities sa
                 LEFT JOIN spend_venues_profiles sp ON (sa.user_id = sp.user_id)
                 LEFT JOIN earn_groups_profiles eap ON (sa.user_id = eap.user_id)
                 INNER JOIN images i ON (sa.image = i.id)
+                WHERE sa.is_live = 'true' AND sp.is_public = 'true'
```
Comment (RamyAlshurafa) [inline server/modules/activity/model/find.js:28]:
```suggestion
                WHERE sa.is_live IS true  AND (sp.is_public IS true OR eap.is_public IS true)
```

to get the spend activities been made by the earn groups

File: server/modules/activity/model/find.js [hunk 3]
```diff
@@ -3,30 +3,29 @@ import { query } from '../../../database';
...
-                  i.key "activity.image_key",
-                  i.bucket "activity.image_bucket",
-                  i.bucket_region "activity.image_bucket_region",
+                  sa.id "activity_id",
+                  sa.name "activity_name",
+                  sa.sub_title "subtitle",
+                  sa.person_tc_price "cost",
+                  sa.activity_types::VARCHAR[] "activity_types",
+                  sa.is_online "online",
+                  sp.city "townOrCity",
```
Comment (RamyAlshurafa) [inline server/modules/activity/model/find.js:12]: the address not displaying when it created by earn group
image

**Comment by thejoefriel** [comment]
Note, I've created an issue for the menu bug that Ramy has noticed, so that doesn't need to be fixed in this PR

**Pull request merged**